react-validated-input-component/src/validated-input.jsx

```diff
@@ -0,0 +1,56 @@
+import React from "react";
+
+export default class ValidatedInput extends React.Component {
+  constructor(props){
+    super(props);
+    this.state = {
+      password: ""
+    };
+    this.handleChange = this.handleChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  handleChange(event){
+    this.setState({password: event.target.value})
+  }
+
+  handleSubmit(event){
+    event.preventDefault();
+  }
+
+  render(){
+    const input = this.state.password;
+    if (!input) {
+      return (
+        <form onSubmit={this.handleSubmit}>
+          <h2>Password</h2>
+          <input type="text" value={this.state.password} onChange={this.handleChange}/>
+          <span>
+            <i className="x-cross" class="fas fa-user-times"></i>
+          </span>
+          <p className="message">A password is required</p>
+        </form>
+      );
+    } else if ( input.length < 8){
+      return (
+        <form onSubmit={this.handleSubmit}>
+          <h2>Password</h2>
+          <input type="text" value={this.state.password} onChange={this.handleChange} />
+          <span>
+            <i className="x-cross" class="fas fa-user-times"></i>
+          </span>
+          <p className="message">Your password is too short</p>
+        </form>
+      );
+    } else {
+      return (
+        <form onSubmit={this.handleSubmit}>
+          <h2>Password</h2>
+          <input type="text" value={this.state.password} onChange={this.handleChange} />
+          <span>
+            <i class="fas fa-user-check"></i>
+          </span>
+        </form>);
+    }
```

Comment:
We want to avoid recreating the whole form 3 separate times. There should be only one `return` statement here with dynamically setting the parts that need to change. You can use an if statement to determine the `icon` to be used and the `message` to display. For example you can do something like:

```JSX
let icon = 'fa-user-check';
let message = '';

if (!input) {
  icon = 'fa-user-times';
  message = 'A password is required'
} else if (input.length < 8) {
  icon = 'fa-user-times';
  message = 'Your password is too short';
}

return (
  <form onSubmit={this.handleSubmit}>
    <h2>Password</h2>
    <input type="text" value={this.state.password} onChange={this.handleChange} />
    <span>
      <i className="x-cross" class={`fas ${icon}`}></i>
    </span>
    <p className="message">{message}</p>
  </form>
);
```

The goal is to reduce duplication of effort and to make the code easy to maintain. You don't want to have to make changes in 3 separate places.